tests/schema/test_union.py

```diff
@@ -498,6 +500,35 @@ def animal(self) -> animal_union | None:
     assert result.data["animal"] is None
 
 
+@pytest.mark.skipif(
+    sys.version_info < (3, 10),
+    reason="pipe syntax for union is only available on python 3.10+",
+)
+def test_union_optional_scalar_with_or_operator():
```

Comment:
could you move this test to the scalars tests? 😊

even though it using the union syntax, it is not really a GraphQL union 😊

Reply:
done